packages/desktop/components/popup/popups/SignTypedDataPopup.svelte

```diff
@@ -0,0 +1,105 @@
+<script lang="ts">
+    import { localize } from '@core/i18n'
+    import { PopupId, closePopup, openPopup } from '@desktop/auxiliary/popup'
+    import { handleError } from '@core/error/handlers'
+    import { IConnectedDapp } from '@auxiliary/wallet-connect/interface'
+    import { CallbackParameters } from '@auxiliary/wallet-connect/types'
+    import { Alert, JsonTree, Table, Text } from '@bloomwalletio/ui'
+    import { IAccountState } from '@core/account'
+    import { selectedAccount } from '@core/account/stores'
+    import { IChain } from '@core/network'
+    import { AccountLabel } from '@ui'
+    import { checkActiveProfileAuthAsync } from '@core/profile/actions'
+    import { LedgerAppName } from '@core/ledger'
+    import PopupTemplate from '../PopupTemplate.svelte'
+    import DappDataBanner from '@components/DappDataBanner.svelte'
+    import { SignTypedDataVersion } from '@metamask/eth-sig-util'
+    import { signEip712Message } from '@core/wallet/actions/signEip712Message'
+
+    export let data: string
+    export let version: SignTypedDataVersion.V3 | SignTypedDataVersion.V4
+    export let account: IAccountState
+    export let chain: IChain
+    export let dapp: IConnectedDapp | undefined
+    export let callback: (params: CallbackParameters) => void
+
+    let isBusy = false
+
+    async function unlockAndSign(): Promise<string> {
+        await checkActiveProfileAuthAsync(LedgerAppName.Ethereum)
+        const { coinType } = chain.getConfiguration()
+        return await signEip712Message(data, version, coinType, account)
+    }
+
+    async function onConfirmClick(): Promise<void> {
+        isBusy = true
+        try {
+            const result = await unlockAndSign()
+            closePopup({ forceClose: true })
+
+            callback({ result })
+            openPopup({
+                id: PopupId.SuccessfulDappInteraction,
+                props: {
+                    successMessage: localize('popups.signMessage.success'),
+                    url: dapp.metadata?.url,
+                },
+            })
+        } catch (err) {
+            closePopup({ forceClose: true })
+            callback({ error: err ?? localize('error.global.generic') })
+            handleError(err)
+        } finally {
+            isBusy = false
+        }
+    }
+
+    function onCancelClick(): void {
+        closePopup({ callOnCancel: true })
+    }
+</script>
+
+<PopupTemplate
+    title={localize('popups.signMessage.title')}
+    backButton={{
+        text: localize('actions.cancel'),
+        onClick: onCancelClick,
+    }}
+    continueButton={{
+        text: localize('popups.signMessage.action'),
+        onClick: onConfirmClick,
+    }}
+    busy={$selectedAccount?.isTransferring || isBusy}
```

Comment:
```suggestion
    busy={isBusy}
```
we never set `isTransferring`